doc/modules.md

```diff
@@ -0,0 +1,96 @@
+Modularization
+============
+
+This document describes the changes for a more fine grained modularization
+of common libraries, e.g., Lunchbox
+
+## Requirements
+
+* Per-functionality targets
+* Improved build times for super projects
+
+## Implementation
+
+* All sub project targets are excluded from the default build, and
+  required targets are built only based on the dependencies.
+* Sub-libraries have a top-level header, e.g., lunchbox/log.h, and a
+  sub-directory with the implementation and CMakeLists, e.g.,
+  lunchbox/log/log.h and .cpp. The new common_add_library() cmake module
+  auto-generates the top-level header. New sub libraries set
+  NAME_OMIT_LIBRARY_HEADER.
+* The old common_library is deprecated, and stays during the transition
+  period.
+* Optional sub-targets are to be avoided and hidden by facades (esp. in
+  Lunchbox, see PersistentMap). If they are necessary, e.g, in hwsd,
+  downstream code needs to test for the target's availability (as done
+  e.g. in eq/server).
+* Servus will eventually be reintegrated as a lunchbox sub library.
+
+## Naming
+
+    common_add_library(Lunchbox::Log HEADERS logImpl.h PUBLIC_HEADERS log.h
+      SOURCES log.cpp DEPENDS PUBLIC Lunchbox::Bar PRIVATE Lunchbox::Foo)
```

Comment:
thanks @tribal-tec for the clarification, althogh I am not sure I understand the docuementation correcly. So inside the Lunchbox project, are the targets named differently (wihtout the ::) because it is not an imported target? So there is LunchboxLog depends on LunchboxAny instead of Lunchbox::Any, for instance?